platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/net/ConnectivityReceiver.java

```diff
@@ -10,78 +10,64 @@
 
 import com.mapbox.mapboxsdk.Mapbox;
 
-import java.util.List;
-import java.util.concurrent.CopyOnWriteArrayList;
-
-import timber.log.Timber;
-
 /**
  * Interface definition for a callback to be invoked when connectivity changes.
  * Not public api.
  */
 public class ConnectivityReceiver extends BroadcastReceiver {
   private static ConnectivityReceiver INSTANCE;
+  private final Context context;
+  private final NativeConnectivityListener nativeConnectivityListener = new NativeConnectivityListener();
+  private int counter;
 
   /**
    * Get or create the singleton instance
    */
   public static synchronized ConnectivityReceiver instance(Context context) {
     if (INSTANCE == null) {
-      // Register new instance
-      INSTANCE = new ConnectivityReceiver();
-      context.registerReceiver(INSTANCE, new IntentFilter("android.net.conn.CONNECTIVITY_CHANGE"));
-
-      // Add default listeners
-      INSTANCE.addListener(new NativeConnectivityListener());
+      INSTANCE = new ConnectivityReceiver(context);
     }
-
     return INSTANCE;
   }
 
-  private List<ConnectivityListener> listeners = new CopyOnWriteArrayList<>();
```

Comment:
If we don't have a list, the indirection is no longer needed (`ConnectivityListener` interface and `NativeConnectivityListener` implementation)